exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py

```diff
@@ -142,50 +153,144 @@ def export(
+            bound_str = "+Inf" if bound == float("inf") else str(bound)
+            value = histogram_record.aggregator.checkpoint[bound]
+            timeseries.append(
+                self._create_timeseries(
+                    histogram_record,
+                    histogram_record.instrument.name,
+                    value,
+                    labels=[("le", bound_str)],
+                )
+            )
+            count += value
+        name = histogram_record.instrument.name + "_count"
+        timeseries.append(
+            self._create_timeseries(histogram_record, name, float(count))
+        )
+        return timeseries
+
+    def _convert_from_last_value(
         self, last_value_record: ExportRecord
     ) -> TimeSeries:
-        raise NotImplementedError()
-
-    def convert_from_value_observer(
+        return [
+            self._create_timeseries(
+                last_value_record,
+                last_value_record.instrument.name,
+                last_value_record.aggregator.checkpoint,
+            )
+        ]
+
+    def _convert_from_value_observer(
         self, value_observer_record: ExportRecord
     ) -> TimeSeries:
-        raise NotImplementedError()
+        timeseries = []
+        for agg_type in ["min", "max", "sum", "count", "last"]:
+            timeseries.append(
+                self._create_timeseries(
+                    value_observer_record,
+                    value_observer_record.instrument.name + "_" + agg_type,
+                    getattr(
+                        value_observer_record.aggregator.checkpoint, agg_type
+                    ),
+                )
+            )
+        return timeseries
 
-    def convert_from_quantile(
+    # TODO: Implement convert from quantile once supported by SDK for Prometheus Summaries
+    def _convert_from_quantile(
         self, summary_record: ExportRecord
     ) -> TimeSeries:
         raise NotImplementedError()
 
     # pylint: disable=no-member
-    def create_timeseries(
-        self, export_record: ExportRecord, name, value: float
+    def _create_timeseries(
+        self, export_record: ExportRecord, name, value: float, labels=None
     ) -> TimeSeries:
-        raise NotImplementedError()
-
-    def create_sample(self, timestamp: int, value: float) -> Sample:
-        raise NotImplementedError()
-
-    def create_label(self, name: str, value: str) -> Label:
-        raise NotImplementedError()
+        timeseries = TimeSeries()
+        seen = set()
+
+        def add_label(label_name, label_value):
+            # Label name must contain only alphanumeric characters and underscores
+            label_name = re.sub("[^\\w_]", "_", label_name)
+            if label_name not in seen:
+                label = Label()
+                label.name = label_name
+                label.value = label_value
+                timeseries.labels.append(label)
+                seen.add(label_name)
+
+        add_label("__name__", name)
```

Comment:
What's this for?

Reply:
Each TimeSeries requires the "__name__" property to be set. It is what will appear as the metric name when querying the TimeSeries (on Grafana for example).

Reply:
Could you add a comment for that?

Reply:
Added